types/index.d.ts

```diff
@@ -1,21 +1,84 @@
+/**
+ * A database transaction
+ */
 export type Transaction = Array<object>;
+/**
+ * The Schema is a way to to simplify relational queries at query time.
+ * Presently, it allows you to define relationships and uniqueness constraints.
+ */
 export type Schema = object;
+/**
+ * Homebase configuration.
+ */
 export type config = {schema?: Schema, initialData?: Transaction };
+/**
+ * Entities are JSON objects with extra features.
+ */
 export type Entity = {
-  get: (...params:string[]) => any;
+  /**
+   * Get a value from the `Entity` given its attribute(s).
+   * You can traverse arbitrarily deep relationships by adding nested attributes to the attributes array.
+   * @param attributes a string array of parameter names
+   */
+  get: (...attributes:string[]) => any;
 }
+/**
+ * Facts are richer key-value pairs that represent your data.
+ */
 export type Datom = [number, string, string | number | object | Array<any>, number, boolean];
+/**
+ * The homebaseClient object allows you to interact with the backend database
+ */
 export type homebaseClient = {
+  /**
+   * Serializes the whole db including the schema to a string
```

Comment:
dbToString and dbToDatoms could use @returns https://jsdoc.app/tags-returns.html